dev.c

```diff
@@ -222,24 +231,15 @@ static void request_end(struct fuse_conn *fc, struct fuse_req *req,
 	if (shouldfree) fuse_request_free(req);
 }
 
-void fuse_request_send_nowait(struct fuse_conn *fc, struct fuse_req *req, bool force)
+void fuse_request_send_nowait(struct fuse_conn *fc, struct fuse_req *req)
 {
-	req->in.unique = fuse_get_unique(fc);
-	fc->request_map[req->in.unique & (FUSE_MAX_REQUEST_IDS - 1)] = req;
-
 	/*
 	 * Ensures checking the value of allow_disconnected and adding request to
 	 * queue is done atomically.
 	 */
 	rcu_read_lock();
 
-	if (force) {
-		queue_request(fc, req);
-		if (fc->connected || fc->allow_disconnected) {
-			fuse_conn_wakeup(fc);
-		}
-		rcu_read_unlock();
-	} else if (fc->connected || fc->allow_disconnected) {
+	if (READ_ONCE(fc->connected) || READ_ONCE(fc->allow_disconnected)) {
```

Comment:
modified all connected and allow_disconnected to use READ/WRITE_ONCE macros so no compiler optimizations/reordering happens.